packages/python/plotly/plotly/figure_factory/_annotated_heatmap.py

```diff
@@ -264,34 +262,48 @@ def get_text_color(self):
             max_text_color = black
         return min_text_color, max_text_color
 
-    def get_z_mid(self):
+    def get_z_min_max(self):
         """
-        Get the mid value of z matrix
+        Get the min and max value of z matrix
 
-        :rtype (float) z_avg: average val from z matrix
+        :rtype (tuple): min and max val from z matrix
         """
         if np and isinstance(self.z, np.ndarray):
-            z_min = np.amin(self.z)
-            z_max = np.amax(self.z)
+            z_min = np.nanmin(self.z)
+            z_max = np.nanmax(self.z)
         else:
             z_min = min([v for row in self.z for v in row])
             z_max = max([v for row in self.z for v in row])
-        z_mid = (z_max + z_min) / 2
-        return z_mid
+        return z_min, z_max
 
-    def make_annotations(self):
+    def make_annotations(self, zmin=None, zmid=None, zmax=None):
         """
         Get annotations for each cell of the heatmap with graph_objs.Annotation
 
         :rtype (list[dict]) annotations: list of annotations for each cell of
             the heatmap
         """
         min_text_color, max_text_color = _AnnotatedHeatmap.get_text_color(self)
-        z_mid = _AnnotatedHeatmap.get_z_mid(self)
+        if zmin is None or zmax is None:
+            zmin, zmax = _AnnotatedHeatmap.get_z_min_max(self)
+        if zmid is None:
+            zmid = (zmax + zmin) / 2
+        if min_text_color == max_text_color:
+            # diverging colorscale
+            mid_text_color = "#000000"
```

Comment:
I'm not sure this is a great default... this is assuming the diverging scale is light in the middle?

Reply:
I wonder if we shouldn't accept a third color in `font_colors` for this and only set this one here if it's not present?

Reply:
Yes can be an option but is there any diverging colorscale where the it's not the case?

Reply:
Well, users can pass in their own colorscale, they're not limited to just the built-in ones.